Remove period from "Update draft release notes..." commit message

mokagio opened the pull request:
## What does it do?

The first line of a commit message should not have a period (`.`)

I noticed this while looking at https://github.com/Automattic/pocket-casts-ios/pull/2484 in the context of p1733261974076509-slack-C029BMLUGRX.

I am not aware of any official Git documentation for the statement above. However, it has been a convention in all the projects and companies I worked on.

The style is also present in the example commit message from the official Git book. See
https://git-scm.com/docs/MyFirstContribution#:~:text=You%20will%20be%20presented%20with%20your,A%20U%20Thor%20%3Cauthor%40example.com%3E

The style is also used in other sources, such as:

- https://commit.style/
- https://www.gitkraken.com/learn/git/best-practices/git-commit-message#using-imperative-verb-form
- https://thoughtbot.com/blog/5-useful-tips-for-a-better-commit-message

One could argue that the 50 to 72 character length and the period omission were characters economy constraints from back in the day of small screens and text-only interfaces. While that is true, I find the are many benefits in forcing oneself to be concise in the commit title/summary.

As for short line length, it works well for people like me, who use big font sizes because their eyes strain otherwise. 🤓

But aside from all the justifications above, having commit message titles without a period is the existing convention in this repo, which make this change a no-brainer.


## Checklist before requesting a review

- [x] Run `bundle exec rubocop` to test for code style violations and recommendations
- [x] Add Unit Tests (aka `specs/*_spec.rb`) if applicable
- [x] Run `bundle exec rspec` to run the whole test suite and ensure all your tests pass
- [x] Make sure you added an entry in [the `CHANGELOG.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#trunk) to describe your changes under the appropriate existing `###` subsection of the existing `## Trunk` section.
- [x] If applicable, add an entry in [the `MIGRATION.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/MIGRATION.md) to describe how the changes will affect the migration from the previous major version and what the clients will need to change and consider.

mokagio committed:
Remove period from "Update draft release notes..." commit message

The first line of a commit message should not have a period (`.`)

I am not aware of any official Git documentation for the statement
above. However, it has been a convention in all the projects and
companies I worked on.

The style is also present in the example commit message from the
official Git book. See
https://git-scm.com/docs/MyFirstContribution#:~:text=You%20will%20be%20presented%20with%20your,A%20U%20Thor%20%3Cauthor%40example.com%3E

The style is also used in other sources, such as:

- https://commit.style/
- https://www.gitkraken.com/learn/git/best-practices/git-commit-message#using-imperative-verb-form
- https://thoughtbot.com/blog/5-useful-tips-for-a-better-commit-message

One could argue that the 50 to 72 character length and the period
omission were characters economy constraints from back in the day of
small screens and text-only interfaces. While that is true, I find the
are many benefits in forcing oneself to be concise in the commit
title/summary. As for short line length, it works well for people like
me, who use big font sizes because their eyes strain otherwise.

But aside from all the justifications above, having commit message
titles without a period is the existing convention in this repo, which
make this change a no brainer.

mokagio committed:
Add changelog entry for '.' removal from commit title

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio commented on lib/fastlane/plugin/wpmreleasetoolkit/actions/common/extract_release_notes_for_version_action.rb:
```diff
@@ -41,7 +41,7 @@ def self.extract_notes(release_notes_file_path, version)
       def self.check_and_commit_extracted_notes_file(file_path, version)
         Action.sh("git add #{file_path}")
-        Action.sh("git diff-index --quiet HEAD || git commit -m \"Update draft release notes for #{version}.\"")
+        Action.sh("git diff-index --quiet HEAD || git commit -m \"Update draft release notes for #{version}\"")
```
Aside from the formatting, it would have been good to update the code to use either the Fastlane Git API or the git wrapper gem that I think we already have setup. 

But I'm in the middle of my support rotation and already feel guilty for indulging in this fix.

AliSoftware replied:
Took the liberty to do exactly that in 72b0592d.

Tested by running `extract_release_notes_for_version` in a dummy branch of PCiOS, worked as expected (first run created the commit with changes to the release notes files, second call detected there was nothing new to commit)

AliSoftware committed:
Update git-commit implementation to use fastlane actions

As suggested in https://github.com/wordpress-mobile/release-toolkit/pull/622#discussion_r1868447996
